lib/Sema/CSSimplify.cpp

```diff
@@ -2697,6 +2697,14 @@ ConstraintSystem::matchTypes(Type type1, Type type2, ConstraintKind kind,
     }
   }
 
+  if (type1->getOptionalObjectType() && type2->isAny()) {
+    auto *fix = AllowImplicitCoercionToAny::create(
+        *this, type1, type2, getConstraintLocator(locator));
+    recordFix(fix);
+    return matchTypes(type1->lookThroughAllOptionalTypes(), type2,
```

Comment:
Adding the fix to the conversionsOrFixes array doesn’t result in the fix being applied, so had to call recordFix.

Although there’s definitely an issue with this logic because it’s triggering the warning for a few cases where it shouldn’t be doing that.

Reply:
Since implicit coercions like these are allowed `repairFailures` wouldn't even be triggered, fix should be attached somewhere in https://github.com/apple/swift/blob/master/lib/Sema/CSSimplify.cpp#L2409L2426 . It's okay to use `recordFix` directly in this situation.

Reply:
Okay, done!